components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -530,6 +530,18 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
             }
 
             userNames = paginateUsers(userNames, limit, offset);
+            //When email is used as user name there are two mail attributes in um_user_attribute column there for it creates duplicates as user names #4181
+            if(userNames[0].equals(userNames[1])){//This only happens when user name is given as the email address hence results can have only two elements
```

Comment:
What happens if the array contains more than two values?

Reply:
The username is always unique. The only case it can get duplicated is when email is used as the user name. So if array contain more than two values the if condition will be false.

Reply:
resolved by adding a hashset replacing the if condition